pkg/ingester/label_names_and_values.go

```diff
@@ -167,6 +186,30 @@ func labelValuesCardinality(
 	return nil
 }
 
+func computeLabelValuesSeriesCount(
+	ctx context.Context,
+	lbName string,
+	lbValues []string,
+	idxReader tsdb.IndexReader,
+	matchedPostingsCloner *index.PostingsCloner,
+	countCh chan<- labelValueCountResult,
+) {
+	// Failing job error will be sent over the countCh channel, so no need to handle it here.
+	_ = concurrency.ForEachJob(ctx, len(lbValues), maxConcurrency, func(ctx context.Context, idx int) error {
+		lbValue := lbValues[idx]
+
+		seriesCount, err := countLabelValueSeries(ctx, lbName, lbValue, idxReader, matchedPostingsCloner)
+
+		countCh <- labelValueCountResult{
+			val:   lbValue,
+			count: seriesCount,
+			err:   err,
+		}
+		return err
+	})
+	close(countCh)
```

Comment:
`concurrency.ForEachJob` will return after first error returned, so chan will be closed and next write will panic. 
You need a wait group here to close the channel.